### PR TITLE
Fix compilation on windows

### PR DESCRIPTION
**Pull request by OznOg**
I found several problem when enabling windows compilation.
- use of pthead lib => replaced by c++11 threads
- idem for mutexes
- wrong compilation flags for MS compiler
- replace 'and' by &&

The whole thing does not compile yet, but those patches are needed

CAREFUL: This breaks compilation on Android (until we can use C++11 on Android)
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/162%23issuecomment-132237797%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/162%23issuecomment-132240882%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/162%23discussion_r37306544%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/162%23discussion_r37308301%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/162%23discussion_r37308804%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/162%23issuecomment-132483585%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/162%23issuecomment-132237797%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22You%20breaking%20compilation%20on%20android.%20Better%20add%20it%20in%20to%20the%20pull%20request%20message.%22%2C%20%22created_at%22%3A%20%222015-08-18T14%3A49%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-18T15%3A01%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22Please%20sumbmit%20again%20against%20the%20%5C%22windows_port%5C%22%22%2C%20%22created_at%22%3A%20%222015-08-19T08%3A04%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20d7444309fe5c24dd09bb053a7202bf3d4624f32b%20CMakeLists.txt%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/162%23discussion_r37306544%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Explicitly%20request%20c%2B%2B11%20on%20window%20please.%20This%20will%20help%20detecting%20error%20on%20old%20compilator%20that%20do%20not%20support%20it.%22%2C%20%22created_at%22%3A%20%222015-08-18T14%3A45%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20don%27t%20know%20how%20to%20do%20that%2C%20actually...%22%2C%20%22created_at%22%3A%20%222015-08-18T14%3A58%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-18T15%3A01%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20CMakeLists.txt%3AL41-52%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/162#issuecomment-132237797'>General Comment</a></b>
- <a href='https://github.com/krocard'></a> You breaking compilation on android. Better add it in to the pull request message.
- <a href='https://github.com/krocard'></a> :+1:<a href='#crh-update-none'></a>
- <a href='https://github.com/krocard'></a> Please sumbmit again against the "windows_port"
- [x] <a href='#crh-comment-Pull d7444309fe5c24dd09bb053a7202bf3d4624f32b CMakeLists.txt 6'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/162#discussion_r37306544'>File: CMakeLists.txt:L41-52</a></b>
- <a href='https://github.com/krocard'></a> Explicitly request c++11 on window please. This will help detecting error on old compilator that do not support it.
- <a href='https://github.com/OznOg'></a> I don't know how to do that, actually...

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/162?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/162?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/162'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -41,7 +41,11 @@ execute_process(COMMAND git describe --tags --dirty
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
     OUTPUT_VARIABLE PARAMETER_FRAMEWORK_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall -Wextra")
+if (WIN32)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Wall /WX")
```
Comment (krocard) [inline CMakeLists.txt]: Explicitly request c++11 on window please. This will help detecting error on old compilator that do not support it.

**Comment by krocard** [comment]
You breaking compilation on android. Better add it in to the pull request message.

File: CMakeLists.txt (hunk 1 above)
Comment (OznOg) [inline CMakeLists.txt]: I don't know how to do that, actually...

**Comment by krocard** [comment]
:+1:<a href='#crh-update-none'></a>

File: CMakeLists.txt (hunk 1 above)
Comment (krocard) [inline CMakeLists.txt]: :+1:<a href='#crh-update-complete'></a>

**Comment by krocard** [comment]
Please sumbmit again against the "windows_port"

**Pull request closed without merging**